### DISCUSSION TITLE
ESP32C2 compiling fails with VSC v1.90.2 Tasmota dev 14.1.0.2

### DISCUSSION THREAD

**Discussion by HofeBY** [arendst/Tasmota#21682] [Support request]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
ESP32C2 compiling fails with VSC 1.90.2
Uninstalled VSC; deleted %HOMEPATH%\.platfomio
Reinstall VSC v1.90.2 and download Tasmota dev 14.1.0.2
compling without any configuration changes fails at ESP32C2

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: 14.1.0.2
  - [ ] Pre-compiled
  - [ x] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
 Compiler LOG VSC v1.90.2 with fresh downloaded Tasmota dev

 *  Executing task: C:\Users\Station\.platformio\penv\Scripts\platformio.exe run --environment tasmota32c2 

Processing tasmota32c2 (board: esp32c2; framework: arduino; platform: https://github.com/tasmota/platform-espressif32/releases/download/2024.06.10/platform-espressif32.zip)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
Nothing more to solidify
-DCONFIG_TASMOTA_FLASHMODE_QIO
*** use provided user_config_override.h as planned ***     
*** use provided platformio_override.ini as planned ***    
*** use provided platformio_tasmota_cenv.ini as planned ***
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32c2.html
PLATFORM: Espressif 32 (2024.6.10) > Espressif Generic ESP32-C2 = 4M Flash, Tasmota 2880k Code/OTA, 320k FS
HARDWARE: ESP32C2 120MHz, 272KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.0.1+sha.371d83e
 - tool-esptoolpy @ 4.7.3
 - tool-mklittlefs @ 3.2.0
 - tool-riscv32-esp-elf-gdb @ 11.2.0+20220823
 - tool-xtensa-esp-elf-gdb @ 11.2.0+20230208
 - toolchain-riscv32-esp @ 12.2.0+20230208
Converting tasmota.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ strict
Found 124 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Adafruit CCS811 Library @ 1.0.0  
|-- Adafruit MAX31865 library @ 1.0.1
|-- Adafruit MCP9808 Library @ 1.1.2 
|-- Adafruit PM25 AQI Sensor @ 1.0.6 
|-- Adafruit seesaw Library @ 1.3.1  
|-- Adafruit SGP30 Sensor @ 1.2.0    
|-- Adafruit SGP40 Sensor @ 1.1.0
|-- Adafruit VEML7700 Library @ 1.0.0
|-- ESP8266Audio @ 1.9.5
|-- Berry @ 7.0
|-- Berry animation library for WS2812 leds @ 0.1
|-- Berry custom template @ 0.1
|-- Berry Matter protocol implementation @ 0.1
|-- Berry Tasmota mapping @ 1.0
|-- Berry int64 implementation for 32 bits architecture @ 1.0
|-- HASPmota @ 1.0
|-- Berry mapping to C @ 1.0
|-- BME8563
|-- headers
|-- ccronexpr
|-- WebServer @ 2.0.0
|-- EEPROM 24C128_256_521
|-- ESP32-to-ESP8266-compat @ 0.0.2
|-- FFat @ 2.0.0
|-- FS @ 2.0.0
|-- ams @ 1.0
|-- HttpClient light @ 1.0
|-- I2Cdevlib-Core
|-- LibTeleinfo @ 1.1.7
|-- lvgl @ 9.1.0
|-- Berry mapping to LVGL @ 1.0
|-- arduino-mcp2515-1.0.1
|-- MPU6050 I2C
|-- Mutichannel_Gas_Sensor @ 0.0.1
|-- esp-nimble-cpp @ 1.4.1
|-- NTP Library
|-- OpenTherm Library @ 0.9.0
|-- Process_control @ 1.0.0
|-- re1.5 @ 0.9
|-- ScioSense ENS16x @ 8.0.0
|-- ScioSense ENS210 @ 3.0.0
|-- Sensirion I2C SGP41 @ 0.1.0
|-- SevenSegmentTM1637 @ 1.1.0
|-- SPL06-007 @ 0.1.0
|-- ESP Mail Client @ 3.4.9
|-- LVGL_assets @ 7.0
|-- UdpListener @ 1.0
|-- Unishox Compressor Decompressor highly customized and optimized for ESP8266 and Tasmota @ 1.0
|-- VL53L0X @ 1.0.2
|-- VL53L1X @ 1.0.1
|-- TLS mini client derived from Arduino @ 1.0
|-- Zip-readonly-FS @ 1.0
|-- A4988_Stepper @ 0.0.1
|-- Adafruit Fingerprint Sensor Library @ 2.1.0
|-- Adafruit GFX Library @ 1.5.6
|-- Adafruit LED Backpack Library @ 1.1.6
|-- Adafruit TSL2591
|-- base64 @ 1.1.1
|-- BME68x
|-- DHT sensor library @ 1.4.3
|-- DNSServer @ 2.0.0
|-- DS2480 Library @ 0.1.0
|-- Waveshare esp 2.9 inch e-paper display driver @ 1.0
|-- Waveshare esp 4.2 inch e-paper display driver @ 1.0
|-- Display renderer @ 1.0
|-- ESP KNX IP Library @ 0.5.2
|-- ESP8266SAM @ 1.0.1
|-- ESPFtpServer @ 0.1.0
|-- Ethernet @ 2.0.0
|-- Ext-printf @ 1.0
|-- FrogmoreScd30
|-- FrogmoreScd40
|-- FT5206_Library @ 1.0.0
|-- GT911
|-- HPMA115S0 Arduino Library @ 1.0.0
|-- IRremoteESP8266 @ 2.8.6
|-- JSMN JSON parser customized and optimized for ESP8266 and Tasmota @ 1.0
|-- KeeloqLib @ 1.1
|-- mp3_shine_esp32 @ 1.0.0
|-- LedControl @ 1.0.6
|-- LinkedList
|-- LiquidCrystal_I2C
|-- LittleFS @ 2.0.0
|-- TasmotaLList @ 1.0
|-- LOLIN_HP303B @ 1.0.0
|-- LoRa @ 0.8.0
|-- MFRC522 @ 1.4.7
|-- MLX90640
|-- MPU_accel
|-- NeoPixelBus @ 2.6.7
|-- NewPing @ 1.9.1
|-- OneWire @ 2.3.2
|-- PubSubClient @ 2.8
|-- RA8876 @ 1.0.2
|-- RadioLib @ 6.4.2
|-- rc-switch @ 1.0.0
|-- RF24 @ 1.3.3
|-- SD @ 2.0.0
|-- SD_MMC @ 2.0.0
|-- Sensirion I2C SEN5X @ 0.2.0
|-- SPI @ 2.0.0
|-- TasmotaModbus @ 3.6.0
|-- TasmotaSerial @ 3.6.0
|-- Ticker @ 2.0.0
|-- TM1638plus @ 1.7.0
|-- Joba_Tsl2561 @ 2.0.10
|-- BearSSL @ 0.6
|-- universal display Library @ 0.1
|-- WiFiHelper @ 1.0.0
|-- WiFi @ 2.0.0
|-- Wire @ 2.0.0
|-- XPT2046_Touchscreen
Building in release mode
Compiling .pio\build\tasmota32c2\src\tasmota.ino.cpp.o
Linking .pio\build\tasmota32c2\firmware.elf
lto-wrapper.exe: warning: using serial compilation of 24 LTRANS jobs
lto-wrapper.exe: note: see the '-flto' option documentation for more information
Retrieving maximum program size .pio\build\tasmota32c2\firmware.elf
Checking size .pio\build\tasmota32c2\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [==        ]  21.4% (used 59688 bytes from 278528 bytes)
Flash: [=======   ]  66.2% (used 1951986 bytes from 2949120 bytes)
Building .pio\build\tasmota32c2\firmware.bin
esptool.py v4.7.3
Creating esp32c2 image...
Merged 2 ELF sections
Successfully created esp32c2 image.
esp32_create_combined_bin([".pio\build\tasmota32c2\firmware.bin"], [".pio\build\tasmota32c2\firmware.elf"])
Read partitions from  partitions/esp32_partition_app2880k_fs320k.csv
# Name,     Type,   SubType,   Offset,    Size,   Flags
nvs         data    nvs        0x9000     0x5000
otadata     data    ota        0xe000     0x2000
safeboot    app     factory    0x10000    0xD0000
app0        app     ota_0      0xE0000    0x2D0000
spiffs      data    spiffs     0x3B0000   0x50000
Creating filesystem with content:
No files added -> will NOT create littlefs.bin and NOT overwrite fs partition!
Will download safeboot binary from URL:
http://ota.tasmota.com/tasmota32/release/tasmota32c2-safeboot.bin
bin_map_copy([".pio\build\tasmota32c2\firmware.bin"], [".pio\build\tasmota32c2\firmware.elf"])
*** [.pio\build\tasmota32c2\firmware.bin] .pio\build\tasmota32c2\firmware.factory.bin: No such file or directory
=============================================================================================== [FAILED] Took 121.33 seconds ===============================================================================================

Environment    Status    Duration
-------------  --------  ------------
tasmota32c2    FAILED    00:02:01.331
=========================================================================================== 1 failed, 0 succeeded in 00:02:01.331 ===========================================================================================
 *  The terminal process "C:\Users\Station\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 'tasmota32c2'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
There where changes in Platformio ini files. Check yours against the development branch.
Tasmota does compile fine. Tested with every MCU with every commit in CI

Compile run for C2 was successfull, maybe no Internet connection to download the safeboot firmware?

**Comment by Jason2866** [comment]
With last commit a colorful warning is printed when safeboot firmware is not existing and download was not possible.
